backend/account/custom_auth_middleware.py

```diff
@@ -38,73 +26,19 @@ def __call__(self, request: HttpRequest) -> HttpResponse:
         ):  # Should API Key be in settings or just env alone?
             return self.get_response(request)
 
-        if not AuthenticationPluginRegistry.is_plugin_available():
-            self.local_authentication(request)
-        elif request.COOKIES:
-            self.authenticate_with_cookies(request, tenantAccessiblePublicPath)
-        if request.user and request.session and "user" in request.session:
+        if AuthenticationPluginRegistry.is_plugin_available():
+            auth_service: AuthenticationService = (
+                AuthenticationPluginRegistry.get_plugin()
+            )
+        else:
+            auth_service = AuthenticationService()
+
+        is_authenticated = auth_service.is_authenticated(request)
+
+        if is_authenticated:
             StateStore.set(Common.LOG_EVENTS_ID, request.session.session_key)
             response = self.get_response(request)
             StateStore.clear(Common.LOG_EVENTS_ID)
 
             return response
         return JsonResponse({"message": "Unauthorized"}, status=401)
```

Comment:
Do you remember why this returns a `JsonResponse` instead of a `HttpResponse`?

Reply:
I think JsonResponse is is mandatory in the case of json data. view sets and api_view decorators are handled to send JsonResponse.

https://docs.djangoproject.com/en/5.0/ref/request-response/#jsonresponse-objects
https://docs.datadoghq.com/code_analysis/static_analysis_rules/python-django/http-response-with-json-dumps/